frontend/src/components/Page.test.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+  This file is part of Edgehog.
+
+  Copyright 2023 SECO Mind Srl
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  SPDX-License-Identifier: Apache-2.0
+*/
+
+import { describe, it, expect, vi } from "vitest";
+import { screen } from "@testing-library/react";
+import { userEvent } from "@testing-library/user-event";
+import { renderWithProviders } from "setupTests";
+
+import Page from "./Page";
+
+describe("Header", () => {
+  it("renders title correctly", () => {
+    renderWithProviders(<Page.Header title="Header Title" />);
+
+    const pageTitle = screen.getByRole("heading", { name: "Header Title" });
+    expect(pageTitle).toBeVisible();
+    expect(pageTitle).toHaveTextContent("Header Title");
+  });
+
+  it("does not render title if title not given", () => {
+    renderWithProviders(<Page.Header />);
+
+    const pageTitle = screen.queryByRole("heading", { name: "" });
+    expect(pageTitle).not.toBeInTheDocument();
+  });
+
+  it("renders children correctly", () => {
+    const childText = "Test Children";
+    renderWithProviders(<Page.Header>{childText}</Page.Header>);
+
+    const childElement = screen.getByText(childText);
+    expect(childElement).toBeVisible();
+  });
+});
+
+describe("Main", () => {
+  it("renders children correctly", () => {
+    const childText = "Test Children";
+    renderWithProviders(<Page.Main children={childText} />);
+
+    const childElement = screen.getByRole("main");
+    expect(childElement).toBeVisible();
+  });
+});
+
+describe("LoadingError", () => {
+  it("'try again' button does not render when no function given", () => {
+    renderWithProviders(<Page.LoadingError />);
+
+    const tryAgainButton = screen.queryByRole("button", { name: /Try again/i });
+    expect(tryAgainButton).toBeNull();
+  });
```

Comment:
We can use default exact [TextMatch](https://testing-library.com/docs/queries/about#precision) precision.
```suggestion
  it("does not render 'Try Again' button without onRetry property", () => {
    renderWithProviders(<Page.LoadingError />);

    const tryAgainButton = screen.queryByRole("button", { name: "Try again" });
    expect(tryAgainButton).toBeNull();
  });
```